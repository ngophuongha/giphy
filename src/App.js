import "./App.css";
import { GifPage } from "./pages/GifPage";

function App() {
  return <GifPage />;
}

export default App;
