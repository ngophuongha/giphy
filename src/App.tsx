import styled, { ThemeProvider } from "styled-components";
import theme from "styled-theming";
import { useAppSelector } from "./store/hooks";
import "./App.css";
import { GifPage } from "./pages/gif-page/GifPage";

export const backgroundColor = theme("theme", {
  light: "#fff",
  dark: "#2d2d2d",
});

export const textColor = theme("theme", {
  light: "#000",
  dark: "#fff",
});

const Container = styled.div`
  font-family: sans-serif;
  background-color: ${backgroundColor};
  color: ${textColor};
`;

const App = (): JSX.Element => {
  const theme = useAppSelector((state) => state.theme.theme);
  return (
    <ThemeProvider theme={{ theme: theme }}>
      <Container>
        <GifPage />
      </Container>
    </ThemeProvider>
  );
};

export default App;
