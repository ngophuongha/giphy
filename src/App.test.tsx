import { render } from "@testing-library/react";
import App from "./App";

describe("Index Page", () => {
  it("should match snapshot", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
