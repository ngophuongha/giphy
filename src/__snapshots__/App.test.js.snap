// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render without crashing 1`] = `
<div>
  <div
    class="container"
    style="width: 70%;"
  >
    <nav
      class="border-bottom navbar navbar-expand-lg navbar-light"
    >
      <div
        class="container"
      >
        <a
          class="navbar-brand"
          href="#home"
        >
          <img
            alt=""
            class="d-inline-block align-top"
            height="30"
            src="/giphy-logo-1.svg"
            width="30"
          />
           
          GIPHY
        </a>
        <button
          aria-controls="responsive-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="responsive-navbar-nav"
        >
          <div
            class="me-auto navbar-nav"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="#"
              href="#"
              role="button"
              tabindex="0"
            >
              Reactions
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="#"
              href="#"
              role="button"
              tabindex="0"
            >
              Entertainment
            </a>
          </div>
          <div
            class="navbar-nav"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="#"
              href="#"
              role="button"
              tabindex="0"
            >
              Sign up
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="2"
              href="#"
              role="button"
              tabindex="0"
            >
              Sign in
            </a>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="my-4 input-group"
    >
      <input
        aria-label="gif-sticker-search"
        class="form-control"
        placeholder="Search all the GIFs and Stickers"
      />
      <span
        class="input-group-text"
        id="basic-addon2"
      >
        Search
      </span>
    </div>
    <header
      class="d-flex align-items-center"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-chart-line fa-w-16 "
        data-icon="chart-line"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M496 384H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v336c0 17.67 14.33 32 32 32h464c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16zM464 96H345.94c-21.38 0-32.09 25.85-16.97 40.97l32.4 32.4L288 242.75l-73.37-73.37c-12.5-12.5-32.76-12.5-45.25 0l-68.69 68.69c-6.25 6.25-6.25 16.38 0 22.63l22.62 22.62c6.25 6.25 16.38 6.25 22.63 0L192 237.25l73.37 73.37c12.5 12.5 32.76 12.5 45.25 0l96-96 32.4 32.4c15.12 15.12 40.97 4.41 40.97-16.97V112c.01-8.84-7.15-16-15.99-16z"
          fill="currentColor"
        />
      </svg>
       
      <h3
        class="h4 px-3"
      >
        Trending Gifs
      </h3>
    </header>
    <div
      class="infinite-scroll-component__outerdiv"
    >
      <div
        class="infinite-scroll-component "
        style="height: auto; overflow: auto;"
      >
        <div
          class="g-2 row row-cols-lg-4 row-cols-md-3 row-cols-sm-2"
        />
        <div
          class="spinner-grow text-primary"
          role="status"
        >
          <span
            class="visually-hidden"
          >
            Loading...
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
